shared-libs/rules-engine/src/pouchdb-provider.js

```diff
@@ -82,6 +81,7 @@ const medicPouchProvider = db => {
               reporting_period: docTag,
             };
           }
+          throw err;
```

Comment:
Only real change. Everything else is appeasing sonar complaints for code depth, and fixing them by switching functions to use async/await.